nat/nat.go

```diff
@@ -61,75 +55,72 @@ func NewPortMapper(log logging.Logger, r Router) Mapper {
 	}
 }
 
-// Map sets up port mapping using given protocol, internal and external ports
-// and returns the final port mapped. It returns 0 if mapping failed after the
-// maximun number of retries
-func (dev *Mapper) Map(protocol string, intPort uint16, desc string) (uint16, error) {
-	mappedPort := make(chan uint16)
+// Attempt to establish a PnP connection from extPort (exposed to the internet) to our
+// intPort (where our process is listening).
+func (dev *Mapper) Map(protocol string, intPort, extPort uint16, desc string) {
+	if !dev.r.IsPnP() {
+		return
+	}
+
+	// we attempt a port map, and log an Error if it fails.
+	err := dev.retryMapPort(protocol, intPort, extPort, desc, mapTimeout)
+	if err != nil {
+		dev.log.Error("PnP map failed from external port %d to internal port %d with %s", extPort, intPort, err)
```

Comment:
I think this should be a warning...what do you think?

Reply:
if they wanted to use PnP and they didn't, I think an Error is appropriate..